.github/workflows/release.yaml

```
@@ -0,0 +1,33 @@
+name: Release
+on:
+  release:
+    types: [published]
+
+jobs:
+  release:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    if: github.repository == 'pytest-dev/pytest-timeout'
+
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.12
```

`3.x` is a recent stable Python version and means no need to update this line each year:

```suggestion
          python-version: 3.x
```